dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/ArgumentTypesResolver.java

```diff
@@ -17,49 +17,45 @@
-import com.palantir.conjure.java.lib.SafeLong;
+import com.google.common.primitives.Primitives;
 import com.palantir.dialogue.RequestBody;
+import com.palantir.dialogue.annotations.ParameterSerializer;
 import com.palantir.dialogue.annotations.processor.ArgumentType;
 import com.palantir.dialogue.annotations.processor.ArgumentType.OptionalType;
 import com.palantir.dialogue.annotations.processor.ArgumentTypes;
 import com.palantir.dialogue.annotations.processor.ImmutableOptionalType;
 import com.palantir.logsafe.Preconditions;
-import com.palantir.ri.ResourceIdentifier;
-import com.palantir.tokens.auth.BearerToken;
+import com.palantir.logsafe.SafeArg;
 import com.squareup.javapoet.ClassName;
 import com.squareup.javapoet.TypeName;
-import java.time.OffsetDateTime;
+import java.lang.reflect.Method;
 import java.util.Optional;
 import java.util.OptionalInt;
-import java.util.UUID;
-import java.util.function.Function;
 import javax.lang.model.element.Element;
 import javax.lang.model.element.VariableElement;
 import javax.lang.model.type.TypeMirror;
 
 public final class ArgumentTypesResolver {
 
-    /**
-     * Why not generate this by inspecting the {@link com.palantir.dialogue.PlainSerDe} interface?
-     * If we suddenly decide to provide special serialization for a widely used type, we could break wire
-     * compatibility.
-     */
-    private static final ImmutableMap<TypeName, String> PLAIN_SER_DE_TYPES =
-            ImmutableMap.copyOf(KeyedStream.stream(new ImmutableMap.Builder<Class<?>, String>()
-                            .put(BearerToken.class, "BearerToken")
-                            .put(Boolean.class, "Boolean")
-                            .put(OffsetDateTime.class, "DateTime")
-                            .put(Double.class, "Double")
-                            .put(Integer.class, "Integer")
-                            .put(ResourceIdentifier.class, "Rid")
-                            .put(SafeLong.class, "SafeLong")
-                            .put(String.class, "String")
-                            .put(UUID.class, "Uuid")
-                            .build())
-                    .mapKeys((Function<Class<?>, ClassName>) ClassName::get)
-                    .map(value -> "serialize" + value)
-                    .collectToMap());
+    private static final ImmutableMap<TypeName, String> PARAMETER_SERIALIZER_TYPES;
+
+    static {
+        ImmutableMap.Builder<TypeName, String> builder = ImmutableMap.builder();
+        for (Method method : ParameterSerializer.class.getDeclaredMethods()) {
+            Preconditions.checkArgument(
+                    method.getReturnType().equals(String.class),
+                    "Return type is not String",
+                    SafeArg.of("method", method));
+            Preconditions.checkArgument(
+                    method.getParameterCount() == 1,
+                    "Serializer methods should have a single " + "arg",
+                    SafeArg.of("method", method));
```

Comment:
This feels brittle, I'd instead use a unit test to verify, and if/else here to avoid compilation failures if our assumptions are incorrect in future jvms.

I think we should ignore methods that are non-public, non-static, return non-string, or don't have exactly one parameter.

Reply:
Yep, exactly. There's also some weirdness where lambdas can become special methods on the encapsulating class I encountered here: https://github.com/FasterXML/jackson-databind/pull/2928

Reply:
Also "ParameterSerializer" is an interface, so probably no-non public, non-static, non-string; I would kinda prefer this to keep as is, it will be in a test so happy to break tests.

Reply:
Have a look now, I've made the prod class list them explicitly, and there's a consistency test.